experimental/dds/tree2/src/feature-libraries/optional-field/optionalFieldCodecs.ts

```diff
@@ -17,35 +19,60 @@ export const noChangeCodecFamily: ICodecFamily<0> = makeCodecFamily([[0, unitCod
 
 export const makeOptionalFieldCodecFamily = (
 	childCodec: IJsonCodec<NodeChangeset>,
-): ICodecFamily<OptionalChangeset> => makeCodecFamily([[0, makeOptionalFieldCodec(childCodec)]]);
+	revisionTagCodec: IJsonCodec<RevisionTag, EncodedRevisionTag>,
+): ICodecFamily<OptionalChangeset> =>
+	makeCodecFamily([[0, makeOptionalFieldCodec(childCodec, revisionTagCodec)]]);
 
-const registerIdCodec: IJsonCodec<RegisterId, EncodedRegisterId> = {
-	encode: (registerId: RegisterId) => {
-		if (registerId === "self") {
-			return 0;
-		}
+function makeRegisterIdCodec(
+	revisionTagCodec: IJsonCodec<RevisionTag, EncodedRevisionTag>,
+): IJsonCodec<RegisterId, EncodedRegisterId> {
+	return {
+		encode: (registerId: RegisterId) => {
+			if (registerId === "self") {
+				return 0;
+			}
 
-		return { ...registerId };
-	},
-	decode: (registerId: EncodedRegisterId) => {
-		if (registerId === 0) {
-			return "self";
-		}
+			const encodedRegisterId: EncodedRegisterId = { localId: registerId.localId };
```

Comment:
Thanks for fixing the fact with were using deconstruction during encoding here. :)